mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -315,13 +315,15 @@ class Word2Vec extends Serializable with Logging {
     val syn0Global =
       Array.fill[Float](vocabSize * vectorSize)((initRandom.nextFloat() - 0.5f) / vectorSize)
     val syn1Global = new Array[Float](vocabSize * vectorSize)
+    val bcSyn0Global = sc.broadcast(syn0Global)
```

Comment:
Broadcasting is a good idea, but these values are modified (based on aggregated updates) on each iteration.  You'll need to make a new broadcast variable on each iteration.